scripts/generate-grpc

```diff
@@ -2,15 +2,18 @@
 
 set -e
 
+# if BUILDER_FLAG is set, then the script will exit with the error code
```

Comment:
Why we need to do this? I assume we should always fail with a consistent error code, so the original code looks good to me.

When we need to customize it?

Reply:
Yes, we need to `exit 1` to break this script.
So, in the build flow, we need to take it down if the gRPC check fails.

But to generate the gRPC code and keep the output.
We need to make this script exit normally (which means `exit 0`), so I made the exit variable on-demand.

Reply:
Clear. Sorry for the late reply.

ref:
[/file/file.go#L135-L152](https://github.com/rancher/dapper/blob/c1e5b08924c6430528141534b704546c755cb8f8/file/file.go#L135-L152)
```go
	source := d.env.Source()
	output := d.env.Output()
	if !d.IsBind() && !d.NoOut {
		for _, i := range output {
			p := i
			if !strings.HasPrefix(p, "/") {
				p = path.Join(source, i)
			}
			targetDir := path.Dir(i)
			if err := os.MkdirAll(targetDir, 0755); err != nil {
				return err
			}
			logrus.Infof("docker cp %s %s", p, targetDir)
			if err := d.exec("cp", name+":"+p, targetDir); err != nil {
				logrus.Debugf("Error copying back '%s': %s", i, err)
			}
		}
	}
```